lib/galaxy/managers/markdown_util.py

```diff
@@ -66,71 +237,178 @@ def ready_galaxy_markdown_for_export(trans, internal_galaxy_markdown):
+
+        def walk_elements(collection, element_prefix=""):
+            if ":" in collection.collection_type:
+                for element in collection.elements:
+                    walk_elements(element.child_collection, element_prefix + element.element_identifier + ":")
+            else:
+                for element in collection.elements:
+                    markdown_wrapper[0] += "**Element:** %s%s\n\n" % (element_prefix, element.element_identifier)
+                    markdown_wrapper[0] += self._display_dataset_content(element.hda, header="Element Contents")
+        walk_elements(hdca.collection)
+        markdown = '---\n%s\n---\n' % markdown_wrapper[0]
+        return (markdown, True)
+
+    def handle_tool_stdout(self, line, job):
+        stdout = job.tool_stdout or "*No Standard Output Available*"
+        return ("**Standard Output:** %s" % stdout, True)
+
+    def handle_tool_stderr(self, line, job):
+        stderr = job.tool_stderr or "*No Standard Error Available*"
+        return ("**Standard Error:** %s" % stderr, True)
+
+    def handle_job_metrics(self, line, job):
+        return ("**Job Metrics**: this isn't implemented for PDF export yet", True)
+
+    def handle_job_parameters(self, line, job):
+        return ("**sJob Parameters**: - this isn't implemented for PDF export yet", True)
+
+
+class MarkdownFormatHelpers(object):
+    """Inject common markdown formatting helpers for per-datatype rendering."""
+
+    @staticmethod
+    def literal_via_fence(content):
+        return "\n%s\n" % "\n".join(["    %s" % l for l in content.splitlines()])
+
+    @staticmethod
+    def indicate_data_truncated():
+        return "\n**Warning:** The above data has been truncated to be embedded in this document.\n\n"
+
+    @staticmethod
+    def pre_formatted_contents(markdown):
+        return "<pre>%s</pre>" % markdown
+
+
+def to_basic_markdown(trans, internal_galaxy_markdown):
+    """Replace Galaxy Markdown extensions with plain Markdown for PDF/HTML export.
+    """
+    markdown_formatting_helpers = MarkdownFormatHelpers()
+    directive_handler = ToBasicMarkdownDirectiveHandler(trans, markdown_formatting_helpers)
+    plain_markdown = directive_handler.walk(trans, internal_galaxy_markdown)
+    return plain_markdown
+
+
+def to_html(basic_markdown):
+    # Allow data: urls so we can embed images.
+    html = sanitize_html(markdown.markdown(basic_markdown, extensions=["tables"]), allow_data_urls=True)
+    return html
+
+
+def to_pdf(trans, basic_markdown, css_paths=[]):
+    as_html = to_html(basic_markdown)
+    directory = tempfile.mkdtemp('gxmarkdown')
+    index = os.path.join(directory, "index.html")
+    try:
+        output_file = codecs.open(index, "w", encoding="utf-8", errors="xmlcharrefreplace")
+        output_file.write(as_html)
+        output_file.close()
+        html = HTML(filename=index)
+        stylesheets = [CSS(string=pkg_resources.resource_string(__name__, 'markdown_export_base.css'))]
```

Comment:
This file doesn't exist for me, preventing generation of the pdf.

Reply:
Missing git add. I'll push it in the next rebase - thanks for the catch.